src/ert/_c_wrappers/job_queue/ext_job.py

```diff
@@ -1,263 +1,211 @@
+        if os.path.isdir(resolved):
+            raise ConfigValidationError(
+                f"ExtJob {name} has executable set to directory {resolved}"
+            )
 
-        if self.get_stdout_file() != other.get_stdout_file():
-            return False
+        return resolved
 
-        return True
+    def copy(self):
+        return replace(self)
+
+    _int_keywords = ["MAX_RUNNING", "MAX_RUNNING_MINUTES", "MIN_ARG", "MAX_ARG"]
+    _str_keywords = [
+        "STDIN",
+        "STDOUT",
+        "STDERR",
+        "START_FILE",
+        "EXECUTABLE",
+        "PORTABLE_EXE",
+        "TARGET_FILE",
+        "ERROR_FILE",
+        "START_FILE",
+    ]
+
+    @classmethod
+    def _parse_config_file(cls, config_file: str):
+        parser = ConfigParser()
+        for int_key in cls._int_keywords:
+            parser.add(int_key, value_type=ContentTypeEnum.CONFIG_INT).set_argc_minmax(
+                1, 1
+            )
+        for path_key in cls._str_keywords:
+            parser.add(path_key).set_argc_minmax(1, 1)
+
+        parser.add("EXECUTABLE", required=True).set_argc_minmax(1, 1)
+        parser.add("ENV").set_argc_minmax(1, 2)
+        parser.add("EXEC_ENV").set_argc_minmax(1, 2)
+        parser.add("DEFAULT").set_argc_minmax(2, 2)
+        parser.add("ARGLIST").set_argc_minmax(1, -1)
+        schema = parser.add("ARG_TYPE")
+        schema.set_argc_minmax(2, 2)
+        schema.iset_type(0, ContentTypeEnum.CONFIG_INT)
+
+        return parser.parse(
+            config_file,
+        )
+
+    @classmethod
+    def _read_str_keywords(cls, content_dict, config_content):
+        def set_nullable(keyword, key):
+            value = config_content.getValue(keyword)
+            if value == "null":
+                value = None
+            content_dict[key] = value
+
+        for key in cls._str_keywords:
+            if config_content.hasKey(key):
+                if key in ("STDIN", "STDOUT", "STDERR"):
+                    set_nullable(key, key.lower() + "_file")
+                else:
+                    set_nullable(key, key.lower())
+
+    @classmethod
+    def _read_int_keywords(cls, content_dict, config_content):
+        for key in cls._int_keywords:
+            if config_content.hasKey(key):
+                value = config_content.getValue(key)
+                if value > 0:
+                    # less than 0 in the config is equivalent to setting None
+                    # (backwards compatability)
+                    content_dict[key.lower()] = value
+
+    @classmethod
+    def from_config_file(cls, config_file: str, name: Optional[str] = None):
+        if name is None:
+            name = os.path.basename(config_file)
+
+        try:
+            config_content = cls._parse_config_file(config_file)
+        except IOError as err:
+            raise ConfigValidationError(
+                f"Could not open job config file {config_file}"
+            ) from err
+
+        content_dict = {}
```

Comment:
It was a bit difficult to follow what this `content_dict` actually does I think I understand it but I also think that it would have been shorter and more clear if you just used the actually variables for the constructor like:

```python
    stdin_file: Optional[str] = config_content.get("STDIN")
    stdout_file: Optional[str] = config_content.get("STDOUT")
    stderr_file: Optional[str] = config_content.get("STDERR")
```
and implement the get inside the config_content.py something like: 

``` python
    def get(self, key):
        if not self.hasKey(key):
            return None
        value = self[key].getValue(-1, 0)
        if value == "null":
            return None
        return value
```
But I guess it is a matter of style

Reply:
I think that might be premature as these are the only field that interprets "null" as None.